sql/core/src/test/scala/org/apache/spark/sql/streaming/WatermarkSuite.scala

```diff
@@ -96,28 +96,58 @@ class WatermarkSuite extends StreamTest with BeforeAndAfter with Logging {
     )
   }
 
-  ignore("recovery") {
-    val inputData = MemoryStream[Int]
-
-    val windowedAggregation = inputData.toDF()
+  test("recovery") {
+    val ms = new MemoryStream[Int](0, sqlContext)
+    val df = ms.toDF().toDF("a")
+    val tableName = "recovery"
+    def startQuery: StreamingQuery = {
+      ms.toDF()
         .withColumn("eventTime", $"value".cast("timestamp"))
         .withWatermark("eventTime", "10 seconds")
         .groupBy(window($"eventTime", "5 seconds") as 'window)
         .agg(count("*") as 'count)
         .select($"window".getField("start").cast("long").as[Long], $"count".as[Long])
+        .writeStream
+        .format("memory")
+        .queryName(tableName)
```

Comment:
i think this is not testing it correctly! you are not even setting checkpoint directory. so offset log is not recovered.